tests/integration/security/authorization_test.go

```diff
@@ -494,46 +499,52 @@ func TestAuthorization_NegativeMatch(t *testing.T) {
 					// a connects to b, c and d in ns1 with mTLS.
 					// bInNs2 connects to b and c with mTLS, to d with plain-text.
 					cases := []func(testContext framework.TestContext){
-						// Test the policy with overlapped `paths` and `not_paths` on b.
+						// Test the policy with overlapped `paths`, `not_paths` and `not_methods` on b.
 						// a and bInNs2 should have the same results:
 						// - path with prefix `/prefix` should be denied explicitly.
 						// - path `/prefix/allowlist` should be excluded from the deny.
 						// - path `/allow` should be allowed implicitly.
-						newTestCase(a[0], b, "/prefix", false),
-						newTestCase(a[0], b, "/prefix/other", false),
-						newTestCase(a[0], b, "/prefix/allowlist", true),
-						newTestCase(a[0], b, "/allow", true),
-						newTestCase(bInNS2[0], b, "/prefix", false),
-						newTestCase(bInNS2[0], b, "/prefix/other", false),
-						newTestCase(bInNS2[0], b, "/prefix/allowlist", true),
-						newTestCase(bInNS2[0], b, "/allow", true),
+						newTestCase(a[0], b, "/", false, "GET", "", "http"),
```

Comment:
the argument list of the function is way too long and you only need to customize the method/port/host for 1-2 test cases but need to duplicate the parameter in all other places, this is not good.

Can you refactor to pass a struct to the NewTestCase? For example

```
type struct request{
  method string // default value is GET if not specified
  host string / only set if not empty
  port string // default value is http if not specified
  path string // should always be set
}
```

You only specify the field in the struct if it's different from the default value.